samples/subsys/shell/shell_module/src/main.c

```diff
@@ -145,5 +147,18 @@ SHELL_CMD_ARG_REGISTER(version, NULL, "Show kernel version", cmd_version, 1, 0);
 
 void main(void)
 {
+#if defined(CONFIG_USB_UART_CONSOLE)
+	const struct device *dev;
+	uint32_t dtr = 0;
 
+	dev = device_get_binding(CONFIG_UART_SHELL_ON_DEV_NAME);
```

Comment:
could you use `const struct device *dev = DEVICE_DT_GET(DT_CHOSEN(zephyr_shell_uart));`?

Reply:
That could be misleading.  https://docs.zephyrproject.org/latest/reference/devicetree/api.html specifies that zephyr,shell-uart is used only to supply to default value for `CONFIG_UART_SHELL_ON_DEV_NAME`.  The user can override the actual device through Kconfig.

It could be switched for the sample on an argument that the sample's prj.conf doesn't do that override, but that could confuse a user who finds the Kconfig then doesn't understand why the device selection taken from the sample doesn't work.

Reply:
More relevantly: is it even possible to reference a CDC-ACM UART from devicetree?  I think the answer is "no".

Reply:
> More relevantly: is it even possible to reference a CDC-ACM UART from devicetree? I think the answer is "no".

It is not possible, but something to discuss, I have use case for that.